src/MOIWrapper.jl

```diff
@@ -0,0 +1,397 @@
+        cbcModelFormat.constraint_matrix[mapping.conmap[ci].value,mapping.varmap[term.variable_index].value] += term.coefficient
+    end
+    cbcModelFormat.row_lb[mapping.conmap[ci].value] = s.value - f.constant
+    cbcModelFormat.row_ub[mapping.conmap[ci].value] = s.value - f.constant
+end
+
+function loadConstraint(ci::MOI.ConstraintIndex, cbcModelFormat::CbcModelFormat, mapping::MOIU.IndexMap,
+    f::MOI.ScalarAffineFunction, s::MOI.GreaterThan)
+    for term in f.terms
+        cbcModelFormat.constraint_matrix[mapping.conmap[ci].value,mapping.varmap[term.variable_index].value] += term.coefficient
+    end
+    cbcModelFormat.row_lb[mapping.conmap[ci].value] = s.lower - f.constant
+end
+
+function loadConstraint(ci::MOI.ConstraintIndex, cbcModelFormat::CbcModelFormat, mapping::MOIU.IndexMap,
+    f::MOI.ScalarAffineFunction, s::MOI.LessThan)
+    for term in f.terms
+        cbcModelFormat.constraint_matrix[mapping.conmap[ci].value, mapping.varmap[term.variable_index].value] += term.coefficient
+    end
+    cbcModelFormat.row_ub[mapping.conmap[ci].value] = s.upper - f.constant
+end
+
+function loadConstraint(ci::MOI.ConstraintIndex, cbcModelFormat::CbcModelFormat, mapping::MOIU.IndexMap,
+    f::MOI.ScalarAffineFunction, s::MOI.Interval)
+    for term in f.terms
+        cbcModelFormat.constraint_matrix[mapping.conmap[ci].value, mapping.varmap[term.variable_index].value] += term.coefficient
+    end
+    cbcModelFormat.row_ub[mapping.conmap[ci].value] = s.upper - f.constant
+    cbcModelFormat.row_ub[mapping.conmap[ci].value] = s.lower - f.constant
+end
+
+
+function loadObj(cbcModelFormat::CbcModelFormat, mapping::MOIU.IndexMap,
+    f::MOI.ScalarAffineFunction)
+    for term in f.terms
+        cbcModelFormat.obj[mapping.varmap[term.variable_index].value] += term.coefficient
+    end
+end
+
+
+```
+    Create inner model cbcOptimizer based on abstract model userOptimizer provided by user.
+    Fill the object of type CbcModelFormat:
+        constraint_matrix::AbstractMatrix,
+        col_lb::VecOrNothing,
+        col_ub::VecOrNothing,
+        obj::VecOrNothing,
+        row_lb::VecOrNothing,
+        row_ub::VecOrNothing,
+    These are needed by function loadProblem of CbcCInterface.
+
+```
+function MOI.copy!(cbcOptimizer::CbcOptimizer,
+    userOptimizer::MOI.ModelLike; copynames=false)
+
+    mapping = MOIU.IndexMap()
+
+    nbCols = MOI.get(userOptimizer, MOI.NumberOfVariables())
+    varIndex = MOI.get(userOptimizer, MOI.ListOfVariableIndices())
+    for i in 1:nbCols
+        mapping.varmap[varIndex[i]] = MOI.VariableIndex(i)
+    end
+
+    zeroOneIndices = Vector{Int}(0)
+    integerIndices = Vector{Int}(0)
+    listOfConstraints = MOI.get(userOptimizer, MOI.ListOfConstraints())
+    nbRows = 0;
+    for (F,S) in listOfConstraints
+        (F,S) in SUPPORTED_CONSTRAINTS || return MOI.CopyResult(MOI.CopyUnsupportedConstraint, "Cbc MOI Interface does not support constraints of type " * (F,S) * ".", nothing)
```

Comment:
You might want do do the loop in a separate function so to make it type stable in this separate function like in
https://github.com/JuliaOpt/MathOptInterface.jl/blob/master/src/Utilities/copy.jl#L123-L125